### DISCUSSION TITLE
Vim support

### DISCUSSION THREAD

**Discussion by Timoyoungster** [community/community#19540] [Copilot Conversations] [answered]
Does the copilot also work with normal vim?
And if not, how long until it will? (if it is possible at all)

**Comment by melvincwng** [comment, marked as answer]
@Timoyoungster, GitHub Copilot works with neovim. I don't think it works with normal vim.

Hope this clarifies.

Reference: [GitHub Copilot Docs](https://github.com/features/copilot/)

**Reply by Timoyoungster** [reply]
Ok, thank you. Would it be possible to add support for normal vim in the future though? Or does the copilot depend on features which neovim has and vim doesn’t?

**Reply by melvincwng** [reply]
@Timoyoungster, I'm afraid I may not be the right person to answer that particular question since I'm not a GitHub staff.

But my suggestion is that if you require immediate usage of GitHub Copilot, you should consider switching to either of the following supported IDEs: a) Visual Studio, b) Visual Studio Code, c) Jetbrains IDE, d) Neovim.

You can check for updates on GitHub Copilot docs regularly to see if support for your favourite IDE is available or not.

Hope this clarifies.

**Reply by melvincwng** [reply]
@Timoyoungster, if this reply has been useful, appreciate it if you could 'Mark as answer' to highlight to others that this question has been addressed.

Otherwise, feel free to follow me on GitHub if you would like to stay in touch.

Thank you and have a good day ahead 😊

**Reply by Timoyoungster** [reply]
Ok, thank you very much! I’ll have to wait and see then.

**Reply by melvincwng** [reply]
@Timoyoungster, pleasure in being able to help ;)

**Comment by tpope** [comment]
The latest release supports Vim 9.0.0162 and higher.

**Reply by qiqiboy** [reply]
Hi,  would it be possible to support multi-line virtual text previews in vim?

**Reply by tpope** [reply]
That's exactly what 9.0.0162 supports.

**Reply by qiqiboy** [reply]
> That's exactly what 9.0.0162 supports.

There is a bug that prevents multi-line virtual text from being seen in vim. I made a fix, and it works: https://github.com/github/copilot.vim/pull/19

#### why this bug

When `text_align` is present, the second paramter `col` should be set to `0`.

```vim
" :h prop_add()
text_align	when "text" is present and {col} is zero;
```

**Reply by tpope** [reply]
That worked as intended on 9.0.0162 but subsequent patches changed the behavior. I'll investigate further and fix in the next release.
